server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java

```diff
@@ -116,7 +116,7 @@ public boolean cacheExists(String name) {
    }
 
    private void checkCacheAvailable(String cacheName) {
-      if (!instance.isRunning(cacheName)) {
+      if (!instance.cacheExists(cacheName)) {
```

Comment:
This has to be isRunning to ensure the getCache method doesn't block. Just because it exists doesn't mean it won't block. I think there is something else happening with the blocking failure.

Reply:
Updated.